api/v1alpha1/networkfence_types.go

```diff
@@ -0,0 +1,110 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FencingOperationResult string
+
+const (
+	// OperationResultSucceeded represents the Succeeded operation state.
+	FencingOperationResultSucceeded FencingOperationResult = "Succeeded"
+
+	// OperationResultFailed represents the Failed operation state.
+	FencingOperationResultFailed FencingOperationResult = "Failed"
+)
+
+// NetworkFenceSpec defines the desired state of NetworkFence
+type NetworkFenceSpec struct {
+	// Operation specifes whether fence or unfence operation is to be performed
+	// on the corresponding CIDR blocks.
+	// It's can be set to either "fence" or "unfence".
+	Operation string `json:"operation,omitempty"`
+
+	// +kubebuilder:validation:MinItems=1
+	// Cidr is the list of CIDR blocks, on which the corresponding fence/unfence
+	// operation are performed.
+	Cidr []string `json:"cidr,omitempty"`
+
+	// BackOffLimit specifies the number of retries allowed before marking reclaim
+	// space operation as failed. If not specified, defaults to 6. Maximum allowed
+	// value is 60 and minimum allowed value is 0.
+	// +optional
+	// +kubebuilder:validation:Maximum=60
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default:=6
+	BackoffLimit int32 `json:"backOffLimit"`
+
+	// RetryDeadlineSeconds specifies the duration in seconds relative to the
+	// start time that the operation may be retried; value MUST be positive integer.
+	// If not specified, defaults to 600 seconds. Maximum allowed
+	// value is 1800.
+	// +optional
+	// +kubebuilder:validation:Maximum=1800
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default:=600
+	RetryDeadlineSeconds int64 `json:"retryDeadlineSeconds"`
+}
+
+// NetworkFenceStatus defines the observed state of NetworkFence
+type NetworkFenceStatus struct {
+	// Result indicates the result of NetworkFence operation.
+	Result FencingOperationResult `json:"result,omitempty"`
+
+	// Message contains any message from the NetworkFence operation.
+	Message string `json:"message,omitempty"`
+
+	// Conditions are the list of conditions and their status.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// Retries indicates the number of times the operation is retried.
+	Retries int32 `json:"retries,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+//+kubebuilder:printcolumn:name="Operation",type="string",JSONPath=".spec.operation"
+//+kubebuilder:printcolumn:name="Clients",type="string",JSONPath=".spec.cidr"
```

Comment:
`Clients` is probably not the right name?

Reply:
Updated it to CIDR now